scheduler/stack_test.go

```diff
@@ -207,3 +207,205 @@ func TestServiceStack_Select_BinPack_Overflow(t *testing.T) {
 		t.Fatalf("bad: %#v", met)
 	}
 }
+
+func TestSystemStack_SetNodes(t *testing.T) {
```

Comment:
Maybe there should be a test suite for the Stack interface to share some of this?